src/main/java/com/avengers/gamera/service/UserService.java

```diff
@@ -51,4 +68,38 @@ public UserInfoDto getUserInfo(String email) {
     private User getByEmail(String email) {
         return userRepository.findByEmail(email).orElseThrow(() -> new ResourceNotFoundException("User", email));
     }
+
+    public void addAuthorityToUser(String email, String name) {
+        User user = getByEmail(email);
+        Authority authority = authorityService.getByAuthorityName(name);
+        log.info("Adding authority {} to user {}", email, name);
+        user.getAuthorities().add(authority);
+    }
+
+    public List<UserGetDto> getAllUsers() {
+        return userRepository.findAll().stream().map(userMapper::userToUserGetDto).collect(Collectors.toList());
+    }
+
+    public UserGetDto getUser(Long userId) {
+        return userMapper.userToUserGetDto(findUser(userId));
+    }
+
+    private User findUser(Long userId) {
+        return userRepository.findById(userId).orElseThrow(()-> new ResourceNotFoundException("User", userId));
+    }
+
+    public UserGetDto updateUser(UserPutDto userPutDto, Long userId) {
+        User user = findUser(userId);
+        user.setName(userPutDto.getName());
+        user.setEmail(userPutDto.getEmail());
+        user.setPassword(userPutDto.getPassword());
+        return userMapper.userToUserGetDto(userRepository.save(user));
+    }
+
+    public void deleteUser(Long userId) {
+        userRepository.deleteById(userId);
```

Comment:
we have a isDeleted in user table, set that filed to true instead of deleted completely